test/FunctionTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+require_once '../function.php';
+
+use PHPUnit\Framework\TestCase;
+
+class FunctionTest extends TestCase
+{
+    public function testGenerateRequest_ShouldReturnEverythingIfNoParameter()
+    {
+        $result = generateRequest();
+        $expectedResult = 'SELECT `item`, `category`, `price`, `remaining` FROM `grocery_item`';
+
+        $this->assertEquals($result, $expectedResult);
+    }
+
+    public function testGenerateRequest_ShouldReturnWhereCategoryWithParam()
+    {
+        $filter = 'pantry';
+        $result = generateRequest($filter);
```

Comment:
I would reverse $result and $expectedResult to conform with the setup, execution and assertion pattern

Reply:
yes changed